topdown/tokens.go

```diff
@@ -146,6 +147,44 @@ func builtinJWTVerifyRS256(a ast.Value, b ast.Value) (ast.Value, error) {
 	return ast.Boolean(true), nil
 }
 
+// Implements Secret JWT signature verification
+func builtinJWTVerifySecret(a ast.Value, b ast.Value) (ast.Value, error) {
+
+	// Process the token
+	astToken, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+	token := string(astToken)
+	parts := strings.Split(token, ".")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("encoded JWT must have 3 sections, found %d", len(parts))
+	}
+
+	headerPayload := []byte(strings.Join(parts[:2], "."))
+	sign, err := builtinBase64UrlDecode(ast.String(parts[2]))
+	if err != nil {
+		return nil, err
+	}
+	signature := []byte(sign.(ast.String))
+
+	// Process Secret input
+	astSecret, err := builtins.StringOperand(b, 2)
+	if err != nil {
+		return nil, err
+	}
+	secret := string(astSecret)
+
+	mac := hmac.New(sha256.New, []byte(secret))
+	mac.Write([]byte(headerPayload))
```

Comment:
[nit] `hash.Hash` (returned by `hmac.New`) is an `io.Writer`, and `Write` [could therefore return an error](https://golang.org/pkg/io/#Writer) -- do we care?